gnovm/cmd/gno/clean.go

```diff
@@ -0,0 +1,116 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/gnolang/gno/gnovm/pkg/gnomod"
+	"github.com/gnolang/gno/tm2/pkg/commands"
+	"github.com/gnolang/gno/tm2/pkg/crypto/keys/client"
+)
+
+type cleanCfg struct {
+	dryRun   bool // clean -n flag
+	verbose  bool // clean -x flag
+	modCache bool // clean -modcache flag
+}
+
+func newCleanCmd(io *commands.IO) *commands.Command {
+	cfg := &cleanCfg{}
+
+	return commands.NewCommand(
+		commands.Metadata{
+			Name:       "clean",
+			ShortUsage: "clean [flags]",
+			ShortHelp:  "Removes generated files and cached data",
+		},
+		cfg,
+		func(ctx context.Context, args []string) error {
+			return execClean(cfg, args, io)
+		},
+	)
+}
+
+func (c *cleanCfg) RegisterFlags(fs *flag.FlagSet) {
+	fs.BoolVar(
+		&c.dryRun,
+		"n",
+		false,
+		"print remove commands it would execute, but not run them",
+	)
+
+	fs.BoolVar(
+		&c.verbose,
+		"x",
+		false,
+		"print remove commands as it executes them",
+	)
+
+	fs.BoolVar(
+		&c.modCache,
+		"modcache",
+		false,
+		"remove the entire module download cache",
+	)
+}
+
+func execClean(cfg *cleanCfg, args []string, io *commands.IO) error {
+	if len(args) > 0 {
+		return flag.ErrHelp
+	}
+
+	path, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	modDir, err := gnomod.FindRootDir(path)
+	if err != nil {
+		return errors.New("not a gno module")
+	}
+
+	if path != modDir && (cfg.dryRun || cfg.verbose) {
+		io.Println("cd", modDir)
+	}
+	err = filepath.WalkDir(modDir, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if d.IsDir() {
+			return nil
+		}
+		if !strings.HasSuffix(path, ".gno.gen.go") && !strings.HasSuffix(path, ".gno.gen_test.go") {
+			return nil
+		}
+		if !cfg.dryRun {
+			if err := os.Remove(path); err != nil {
+				return err
+			}
+		}
+		if cfg.dryRun || cfg.verbose {
+			io.Println("rm", strings.TrimPrefix(path, modDir+"/"))
+		}
+
+		return nil
```

Comment:
```suggestion
		if cfg.dryRun {
			return nil
		}
		if err := os.Remove(path); err != nil {
			return err
		}
		if cfg.verbose {
			io.Println("rm", strings.TrimPrefix(path, modDir+"/"))
		}

		return nil
```

Reply:
Not logically the same. We still need to print incase of dryrun

Reply:
Hmm, right, I didn't realize that, thanks for pointing it out. if it is ok, we can print the log before:

```suggestion
		if cfg.verbose ||  cfg.dryRun {
			io.Println("removing", strings.TrimPrefix(path, modDir+"/"))
		}
		if cfg.dryRun {
			return nil
		}
		return os.Remove(path)
```

If not, you can leave it as it is.

Reply:
Thanks for the suggestion. Leaving it as it is for now.